public/app/features/correlations/Forms/TransformationsEditor.tsx

```diff
@@ -0,0 +1,293 @@
+                          </Stack>
+                        }
+                        invalid={!!formState.errors?.config?.transformations?.[index]?.type}
+                        error={formState.errors?.config?.transformations?.[index]?.type?.message}
+                        validationMessageHorizontalOverflow={true}
+                      >
+                        <InputControl
+                          render={({ field: { onChange, ref, ...field } }) => (
+                            <Select
+                              {...field}
+                              onChange={(value) => {
+                                if (!readOnly) {
+                                  const currentValues = getValues().config.transformations[index];
+                                  setKeptVals({
+                                    ...keptVals,
+                                    [index.toString()]: {
+                                      expression: currentValues.expression,
+                                      mapValue: currentValues.mapValue,
+                                    },
+                                  });
+                                  const newValueDetails = getSupportedTransTypeDetails(value.value);
+
+                                  if (newValueDetails.showExpression) {
+                                    setValue(
+                                      `config.transformations.${index}.expression`,
+                                      keptVals[index.toString()]?.expression || ''
+                                    );
+                                  } else {
+                                    setValue(`config.transformations.${index}.expression`, '');
+                                  }
+
+                                  if (newValueDetails.showMapValue) {
+                                    setValue(
+                                      `config.transformations.${index}.mapValue`,
+                                      keptVals[index.toString()]?.mapValue || ''
+                                    );
+                                  } else {
+                                    setValue(`config.transformations.${index}.mapValue`, '');
+                                  }
+
+                                  onChange(value.value);
+                                }
+                              }}
+                              options={transformOptions}
+                              width={25}
+                              aria-label="Type"
+                            />
+                          )}
+                          defaultValue={field.type}
+                          control={control}
+                          name={`config.transformations.${index}.type`}
+                          rules={{ required: { value: true, message: 'Please select a transformation type' } }}
+                        />
+                      </Field>
+                      <Field
```

Comment:
New fields don't have any `aria-label`s, probably worth adding some.

Reply:
Many were added to correspond to the tests but I double checked and made sure all buttons and fields have an aria label